app/modules/knight.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+from app.modules.armour import Armour
+from app.modules.potion import Potion
+from app.modules.weapon import Weapon
+
+
+class Knight:
+    knights = {}    # stores all instances of class
+
+    def __init__(self, name: str, power: int, health: int) -> None:
+        self.name = name
+        self.power = power
+        self.health = health
+        self.armour = []    # stores all instances of armour after applying
+        self.protection = 0     # updates after applying armours and potions
+        self.weapon = None
+        self.potion = None
+        Knight.knights[self.name] = self  # new instance appends to class dict
+
+    def apply_armour(self, features: list) -> None:
+        for item in features:
+            self.armour.append(Armour(item["part"], item["protection"]))
+
+    def apply_weapon(self, weapon_item: dict) -> None:
+        self.weapon = Weapon(weapon_item)
+
+    def apply_potion(self, potion: dict) -> None:
+        if not potion:
+            return None
+        self.potion = Potion(potion["name"], potion["effect"])
+
+    # apply all features if a knight before fight
+    @staticmethod
+    def apply_features(other: Knight) -> None:
+        other.power += other.weapon.damage_power + sum(
+            [0 if not other.potion else other.potion.effect["power"]]
+        )
+        other.health += sum(
+            [0 if not other.potion else other.potion.effect["hp"]]
+        )
+        other.protection = sum(other.armour) + sum(
+            [0 if not other.potion else other.potion.effect["protection"]]
+        )
```

Comment:
DRY 

think about how it can be optimized so that there is less duplication of code